src/sources/util/decoding/bytes.rs

```diff
@@ -0,0 +1,25 @@
+#![deny(missing_docs)]
+
+use super::Decoder;
+use crate::event::Value;
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+
+/// A decoder which wraps the byte frame as-is.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BytesDecoder;
+
+#[typetag::serde(name = "bytes")]
+impl Decoder for BytesDecoder {
```

Comment:
I'm not sure the benefit of having this noop decoder vs just not applying one when there isn't one configured (the null object pattern is cool but I don't think it's benefitting us here).

It also seems like there's a smoother path from `Option<Decoder>` to `Vec<Decoder>` for chaining later.

Reply:
Hmm, since `Decode` is now `Bytes -> Event` I think we can disregard future concerns of chaining for now (chaining might be better done by some inline `Transform` in the future, whereas the current `Decode` implementation optimizes for the n=1 case).

We need to build the event at some point, this is basically restoring the previous behavior by adding a default.

Handling the default case in each site where we use `Decode` would also work, I don't have any hard preferences here.

Reply:
I think the benefit is having a fixed path everywhere that uses a decoder, instead of the uses having to check for a `None` case and doing this as a fallback.